src/potions/variable_meat_potion.ts

```diff
@@ -0,0 +1,147 @@
+import "core-js/modules/es.object.from-entries";
+import {
+  availableAmount,
+  Effect,
+  effectModifier,
+  haveEffect,
+  historicalAge,
+  historicalPrice,
+  Item,
+  mallPrice,
+  numericModifier,
+  print,
+  retrievePrice,
+  use,
+} from "kolmafia";
+import { maxBy } from "libram";
+import { acquire } from "../acquire";
+import { baseMeat, retrieveUntradeablePrice, turnsToNC } from "../lib";
+import { estimatedGarboTurns } from "../turns";
+import { globalOptions } from "../config";
+
+function triangleNumber(b: number, a = 0) {
+  return 0.5 * (b * (b + 1) - a * (a + 1));
+}
+
+export class VariableMeatPotion {
+  potion: Item;
+  effect: Effect;
+  duration: number;
+  softcap: number; // Number of turns to cap out variable bonus
+  meatBonusPerTurn: number; // meat% bonus per turn
+  cappedMeatBonus: number;
+
+  constructor(
+    potion: Item,
+    softcap: number,
+    meatBonusPerTurn: number,
+    duration?: number,
+    effect?: Effect,
+  ) {
+    this.potion = potion;
+    this.effect = effect ?? effectModifier(potion, "Effect");
+    this.duration = duration ?? numericModifier(potion, "Effect Duration");
+    this.softcap = softcap;
+    this.meatBonusPerTurn = meatBonusPerTurn;
+    this.cappedMeatBonus = softcap * meatBonusPerTurn;
+  }
+
+  use(quantity: number): boolean {
+    acquire(
+      quantity,
+      this.potion,
+      (1.2 * retrievePrice(this.potion, quantity)) / quantity,
```

Comment:
@Pantocyclus where does the 1.2 magic number come from? can we make this a named screaming snake case const somewhere?

Reply:
We are averaging the price/value of an item over `quantity`. If the items have different prices over this range, then there exists individual items whose price is above the average.
The `1.2` is there to ensure that we can still buy the item above its average value (since we already bought quantities below its average value, and also determined that the entire bulk purchase makes sense [we don't want to make only partial purchases of variable meat potions])

Reply:
To add on, we also need to be wary of price cliffs in case something goes terribly wrong (e.g. someone buys out the potions after we've done the averaging), so we definitely do not want a factor of, say, `10`.
I can't tell you whether `1.2` is better than `1.3` though